app/javascript/App/components/NewsfeedPage/CoinBody.js

```diff
@@ -1,64 +1,58 @@
-    this.props.setEntityDetails('coin', details)
-  }
-  render() {
-    const { selectCoinDetails, activeEntity, isLoading } = this.props
-    let coin = selectCoinDetails(activeEntity.id)
-    if (!coin && isLoading('coin'))
-      return <LoadingIndicator className="overlay bg-white-70" />
-    coin = coin.toJS()
-    return (
-      <div className="pa4">
-        <div className="flex justify-between items-center">
-          <div className="f4 fw6 flex items-center">
-            {coin.image_url && (
-              <img className="w2e h2e mr3" src={coin.image_url} alt="" />
-            )}
-            {coin.name}
-            <span className="ml2">({coin.symbol})</span>
-          </div>
-          <div>
-            <WatchButton
-              coinID={coin.id}
-              watching={coin.is_being_watched}
-              onChange={this.onWatchCoin}
-            />
-          </div>
+const CoinBody = (props) => {
+  const { selectCoinDetails, activeEntity, isLoading, user } = props
+  let coin = selectCoinDetails(activeEntity.id)
+  if (!coin && isLoading('coin'))
+    return <LoadingIndicator className="overlay bg-white-70" />
+  return (
+    <div className="pa4">
+      <div className="flex justify-between items-center">
+        <a
+          href={`/coins/${coin.get('slug')}`}
+          className="f4 fw6 flex items-center color-inherit"
+        >
+          {coin.get('image_url') && (
+            <img className="w2e h2e mr3" src={coin.get('image_url')} alt="" />
+          )}
+          {coin.get('name')}
+          <span className="ml2">({coin.get('symbol')})</span>
+        </a>
+        <div>
+          <WatchButton {...props} coin={coin} />
         </div>
-        <div className="min-h12e flex items-center justify-center">
-          <div className="tc">
-            <div className="flex items-center">
-              <span className="f2">
-                <Currency>{coin.market_info.price_usd}</Currency>
-              </span>
-              <span className="ml2">
-                <PercentageChange
-                  number={coin.market_info.percent_change_24h}
-                  className="b db"
-                />
-              </span>
-            </div>
-            <div className="dib ph2 pv1 bg-light-gray f6 mt2">
-              Market:&nbsp;
-              {coin.market_info.market_cap_usd}
-            </div>
+      </div>
+      <div className="min-h12e flex items-center justify-center">
+        <div className="tc">
+          <div className="flex items-center">
+            <span className="f2">
+              <Currency>{coin.getIn(['market_info', 'price_usd'])}</Currency>
+            </span>
+            <span className="ml2">
+              <PercentageChange
+                number={coin.getIn(['market_info', 'percent_change_24h'])}
+                className="b db"
+              />
+            </span>
+          </div>
+          <div className="dib ph2 pv1 bg-light-gray f6 mt2">
+            Market:&nbsp;
```

Comment:
do you think it's better to use html entities or template literals here?

Reply:
No strong opinion here. Probably template literals are nicer